lib/Mail/Milter/Authentication/Protocol/SMTP.pm

```diff
@@ -222,6 +222,45 @@ sub command_param {
     return $p;
 }
 
+sub split_command_params {
+    my ($line) = @_;
+    my @params;
+    $line =~ s/^\s+//;
+    while (length($line) > 0) {
+        my $token_end = find_token_end($line) || length($line);
+        push @params, substr($line, 0, $token_end);
+        $line = substr($line, $token_end);
+        $line =~ s/^\s+//;
+    }
+    return @params;
+}
+
+sub find_token_end {
+    my ($input) = @_;
+    my $len = length($input);
+    my @qstack;
+    for (my $i = 0; $i < $len; $i++) {
+        my $c = substr($input, $i, 1);
+        if ($c eq '\\') {
```

Comment:
There is possibly a subtle bug here, where an escaped \ before a legit space will interpret the space as also escaped and split params in the wrong place.

eg

split_command_params('bar=baz path=foo\\bar\\ foo=bar');

will return

$VAR1 = 'bar=baz';
$VAR2 = 'path=foo\\bar\\ foo=bar';

Reply:
Agreed. I was surprised at how different Postfix is from the spec but I thought it would make sense to follow how it does things. However looking at the spec again, it clearly says dquote and backslash are allowed in ESMTP parameters without any special meaning, and even in the path it says backslash serves as an escape character only within a dquoted string. So I updated the patch with 2 changes, 1. Only treat backslash as an escape character in quoted strings (either dquotes or angle brackets, seems like a reasonable middle ground to me), and 2. Split ESMTP params on whitespace only. Let me know what you think.